File: src/main/java/me/dolphago/dto/MemberDto.java
package me.dolphago.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import me.dolphago.domain.Follower;
import me.dolphago.domain.Following;

@ToString
@Getter
@NoArgsConstructor
public class MemberDto {

    private String githubLogin;
    private String url;

    public MemberDto(String githubLogin, String url) {
        this.githubLogin = githubLogin;
        this.url = url;
    }

    public static MemberDto from(String githubLogin, String url) {
        return new MemberDto(githubLogin, url);
    }

    public static Follower toFollowers(MemberDto memberDto) {
        return new Follower(memberDto.getGithubLogin(), memberDto.getUrl());
    }

    public static Following toFollowings(MemberDto memberDto) {
        return new Following(memberDto.getGithubLogin(), memberDto.getUrl());
    }

}

File: src/main/kotlin/me/dolphago/dto/MemberDto.kt
package me.dolphago.dto

import me.dolphago.domain.Follower
import me.dolphago.domain.Following

data class MemberDto(
    val githubLogin: String,
    val url: String
) {
    companion object {
        fun from(githubLogin: String, url: String): MemberDto {
            return MemberDto(githubLogin, url)
        }

        fun toFollowers(memberDto: MemberDto): Follower {
            return Follower(githubLogin = memberDto.githubLogin, url = memberDto.url)
        }

        fun toFollowings(memberDto: MemberDto): Following {
            return Following(githubLogin = memberDto.githubLogin, url = memberDto.url)
        }
    }
}
